app/src/main/java/com/hwang/xsighting/MainActivity.java

```diff
@@ -14,113 +19,193 @@
 import com.google.android.gms.tasks.OnFailureListener;
 import com.google.android.gms.tasks.OnSuccessListener;
 import com.google.android.gms.tasks.Task;
+import com.google.android.material.bottomnavigation.BottomNavigationView;
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.auth.FirebaseUser;
 import com.google.firebase.auth.GetTokenResult;
+import com.google.firebase.firestore.DocumentChange;
 import com.google.firebase.firestore.DocumentReference;
 import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.EventListener;
 import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreException;
+import com.google.firebase.firestore.QuerySnapshot;
+import com.hwang.xsighting.models.Sighting;
 import com.hwang.xsighting.models.User;
 
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 
 public class MainActivity extends AppCompatActivity {
-    private static FirebaseUser user;
-    private static final int RC_SIGN_IN = 482;
-    private static final String TAG = "MainActivity";
-
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-
-        user = FirebaseAuth.getInstance().getCurrentUser();
-            if (user != null) {
-                // User is signed in
-                showName();
-                Task<GetTokenResult> token = user.getIdToken(false);
-                Log.i(TAG, user.toString());
-            } else {
-                // No user signed in, direct them to Login
-                Log.i(TAG, "About to launch sign in");
-                // Choose authentication providers
-                List<AuthUI.IdpConfig> providers = Arrays.asList(
-                      new AuthUI.IdpConfig.EmailBuilder().build());
-
-                // Create and launch sign-in intent
-                startActivityForResult(
-                      AuthUI.getInstance()
-                              .createSignInIntentBuilder()
-                              .setAvailableProviders(providers)
-                              .build(),
-                      RC_SIGN_IN);
-
-                Log.i(TAG, "Sign up intent Sent");
-            }
+  private static FirebaseUser user;
+  private static final int RC_SIGN_IN = 482;
+  private static final String TAG = "MainActivity";
+  private AllSightingsAdapter adapter;
+  private RecyclerView recyclerView;
+  private RecyclerView.LayoutManager layoutManager;
+
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+
+    super.onCreate(savedInstanceState);
+    user = FirebaseAuth.getInstance().getCurrentUser();
+    if (user != null) {
+      // User is signed in
+//      showName();
+      Task<GetTokenResult> token = user.getIdToken(false);

```

Comment:
You don't seem to be using this token for anything; why are you retrieving it?

Reply:
We will eventually use it